pkg/controller/authentication/job.go

```diff
@@ -91,6 +92,13 @@ func generateJobObject(instance *operatorv1alpha1.Authentication, scheme *runtim
 	reqLogger := log.WithValues("Instance.Namespace", instance.Namespace, "Instance.Name", instance.Name)
 	image := shatag.GetImageRef("IM_INITCONTAINER_IMAGE")
 	resources := instance.Spec.ClientRegistration.Resources
+
+	metaLabels := pkgCommon.MergeMap(map[string]string{"app": jobName}, instance.Spec.Labels)
```

Comment:
If the intent is for the `app` label value to be preserved, then the arguments should be reordered.

Reply:
instead of swapping the arguments in this function, In the MergeMap function body i have swapped it, meaning that i have considered mergeMap as a source(instance.Spec.Labels) to add back app label to mergeMap, so that we preserve app label